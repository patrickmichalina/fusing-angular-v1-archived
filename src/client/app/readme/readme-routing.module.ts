import { ReadmeComponent } from './readme.component'
import { NgModule } from '@angular/core'
import { RouterModule } from '@angular/router'

@NgModule({
  imports: [
    RouterModule.forChild([
      {
        path: '',
        component: ReadmeComponent,
        data: {
          meta: {
            title: 'Readme',
            description: 'Readme description'
          }
        }
      }
    ])
  ],
  exports: [RouterModule]
})
export class ReadmeRoutingModule {}
