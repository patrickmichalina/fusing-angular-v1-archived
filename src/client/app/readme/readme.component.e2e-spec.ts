import { baseUrl, browser } from '../../../../tools/test/jest.e2e-setup'

describe('Readme Page', () => {
  it('should have title', async () => {
    expect.assertions(1)

    const page = browser.goto(`${baseUrl}/readme`)

    const text = await page.evaluate(() => document.title).end()

    expect(text).toContain('Readme')
  })
})
