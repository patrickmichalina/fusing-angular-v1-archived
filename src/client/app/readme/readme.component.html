<mat-card [innerHTML]="page$ | async | MarkdownToHtml | pmSanitizeHtml">
</mat-card>