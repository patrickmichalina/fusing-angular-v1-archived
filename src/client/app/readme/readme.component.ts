import {
  ChangeDetectionStrategy,
  Component,
  ViewEncapsulation
} from '@angular/core'
import { HttpClient } from '@angular/common/http'
import { of } from 'rxjs/observable/of'
import { catchError } from 'rxjs/operators'

@Component({
  selector: 'pm-readme',
  templateUrl: './readme.component.html',
  styleUrls: ['./readme.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  encapsulation: ViewEncapsulation.None
})
export class ReadmeComponent {
  constructor(private http: HttpClient) {}
  readonly page$ = this.http
    .get('./assets/md/README.md', {
      responseType: 'text'
    })
    .pipe(catchError(err => of('Error loading README.md')))
}
