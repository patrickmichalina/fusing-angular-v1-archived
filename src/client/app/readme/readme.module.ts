import { ReadmeRoutingModule } from './readme-routing.module'
import { ReadmeComponent } from './readme.component'
import { NgModule } from '@angular/core'
import { SharedModule } from '../shared/shared.module'
import { MarkdownToHtmlModule } from 'markdown-to-html-pipe'

@NgModule({
  imports: [ReadmeRoutingModule, SharedModule, MarkdownToHtmlModule],
  declarations: [ReadmeComponent],
  exports: [ReadmeComponent]
})
export class ReadmeModule {}
