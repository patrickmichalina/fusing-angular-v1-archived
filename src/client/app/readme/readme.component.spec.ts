import { ReadmeComponent } from './readme.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { ReadmeModule } from './Readme.module'
import { AppTestingModule } from '../../../testing/app-testing.module'

@Component({
  selector: 'test-component',
  template: '<pm-readme></pm-readme>'
})
class TestComponent {}

describe(ReadmeComponent.name, () => {
  let fixture: ComponentFixture<ReadmeComponent>

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        imports: [AppTestingModule.forRoot(), ReadmeModule],
        declarations: [TestComponent]
      }).compileComponents()
    })
  )

  beforeEach(
    async(() => {
      fixture = TestBed.createComponent(ReadmeComponent)
    })
  )

  afterEach(
    async(() => {
      TestBed.resetTestingModule()
    })
  )

  it(
    'should compile',
    async(() => {
      expect(fixture.nativeElement).toBeDefined()
      expect(fixture).toMatchSnapshot()
    })
  )
})
