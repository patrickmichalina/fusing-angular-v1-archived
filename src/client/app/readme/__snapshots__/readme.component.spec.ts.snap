// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadmeComponent should compile 1`] = `
<pm-readme
  http={[Function HttpClient]}
  page$={[Function Observable]}
>
  <mat-card
    class="mat-card"
  />
</pm-readme>
`;
